enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -584,11 +585,15 @@ func ConvertFileToCOW(ctx context.Context, env environment.Env, filePath string,
 //
 // It allows processes to read/write to the file as if it
 // was memory, as opposed to having to interact with it via I/O file operations.
+//
+// Mmap is concurrency safe
 type Mmap struct {
 	ctx                context.Context
 	env                environment.Env
 	remoteInstanceName string
 
+	mu sync.RWMutex
+
 	Offset int64
 
 	data       []byte
```

Comment:
nit: move the mutex above the fields that it's protecting

(also, `dataDir` can be moved up next to `Offset` probably)

```suggestion
	Offset int64

	mu         sync.RWMutex
	data       []byte
```